README.md

```diff
@@ -30,27 +51,233 @@ npm install --save ipfsd-ctl
+   - `err` - is the error set if spawning the node is unsuccessful
+   - `ipfsd` - is an object with two properties: 
+     - `ctl` - an [ipfs-api](https://github.com/ipfs/js-ipfs-api) instance attached to the newly created ipfs node
+     - `ctrl` - an instance of a daemon controller object
+   
+### IPFS Client (ctl)
+
+> An instance of [ipfs-api](https://github.com/ipfs/js-ipfs-api#api)
+
+This instance is returned for each successfully started IPFS daemon, when either `df.spawn({start: true})` (the default) is called, or `ipfsdCtrl.startDaemon()` is invoked in the case of nodes that were spawned with `df.spawn({start: false})`. 
+
+
+### IPFS Daemon Controller (ctrl)
+
+> The IPFS daemon controller that allows interacting with the spawned IPFS process
+
+#### `apiAddr` (getter) 
+
+> Get the address (multiaddr) of connected IPFS API.
+
+- returns multiaddr
+
+#### `gatewayAddr` (getter)
+
+> Get the address (multiaddr) of connected IPFS HTTP Gateway.
+
+- returns multiaddr
+
+#### `repoPath` (getter)
+
+> Get the current repo path.
+
+- returns string
+
+#### `started` (getter)
+
+>  Is the node started.
+
+- returns boolean
+ 
+#### `init ([initOpts], callback)`
+
+> Initialize a repo.
+
+- `initOpts` (optional) - options object with the following entries
+  - `keysize` (default 2048) - The bit size of the identiy key.
+  - `directory` (default IPFS_PATH) - The location of the repo.
+  - `function (Error, Node)` callback - receives an instance of this Node on success or an instance of `Error` on failure
+
+
+#### `shutdown (callback)`
+
+> Delete the repo that was being used. If the node was marked as `disposable` this will be called automatically when the process is exited.
+
+- `function(Error)` callback
+
+#### `startDaemon (flags, callback)`
+
+> Start the daemon.
+
+- `flags` - Flags array to be passed to the `ipfs daemon` command.
+- `function(Error, IpfsApi)}` callback - function that receives an instance of `ipfs-api` on success or an instance of `Error` on failure
+
+
+#### `stopDaemon (callback)`
+
+> Stop the daemon.
+
+- `function(Error)` callback - function that receives an instance of `Error` on failure
+
+#### `killProcess (callback)`
+
+> Kill the `ipfs daemon` process.
+
+First `SIGTERM` is sent, after 10.5 seconds `SIGKILL` is sent if the process hasn't exited yet.
+
+- `function()` callback - Called once the process is killed
+
+
+#### `daemonPid ()`
+
+> Get the pid of the `ipfs daemon` process.
+
+- returns the pid number
+
+
+#### `getConfig (key, callback)`
+
+> Call `ipfs config`
+
+If no `key` is passed, the whole config is returned as an object.
+
+- `key` (optional) - A specific config to retrieve.
+- `function(Error, (Object|string)` callback - function that reseives an object or string on success or an `Error` instance on failure
+
+
+#### `setConfig (key, value, callback)`
+
+> Set a config value.
+
+- `key` - the key to set 
+- `value` - the value to set the key to
+- `function(Error)` callback
```

Comment:
`setConfig(config, cb)`? Where `config` is an object with the keys/values to set?

Reply:
Yes to this ^^